### PR TITLE
feat(content-uploader): Implemented partial upload for successful file transfers

### PR DESCRIPTION
**Pull request by boxshrutisingh**
This PR introduces functionality for partial upload support within the ContentUploader component. With this enhancement, successful file transfers trigger the relevant completion callback, leading to smooth handling of uploads. The implementation ensures that uploads proceed seamlessly, even if certain files encounter errors.

**Changes Made**:

- Added a new prop isPartialUploadEnabled to control partial upload functionality.
- Modified ContentUploader.js to handle partial upload cases.
- Implemented logic to trigger completion callback upon successful partial uploads.
- Included tests in ContentUploader.test.js to validate the behavior of partial uploads under different scenarios.

This feature enhances user experience by allowing uploads to continue uninterrupted.

**Use Case**: 
A user drag and drops, or uploads file via browsing.
 Files start getting uploaded in the content uploader. 
 _Happy Path_ : All files get uploaded we get an array of File {}
 _Sad Path_: Some files get errors.
 _Expectation_: For both happy and sad path we should get the array of File {} ( Files which were successfully uploaded ) and the modal should auto close.

### CODE REVIEW

File: src/elements/content-uploader/ContentUploader.js [hunk 1]
```diff
@@ -936,6 +938,15 @@ class ContentUploader extends Component<Props, State> {
         let view = '';
         if ((items && items.length === 0) || allItemsArePending) {
             view = VIEW_UPLOAD_EMPTY;
+        } else if (isPartialUploadEnabled && areAllItemsFinished) {
+            const filesToBeUploaded = items.filter(item => item.status === STATUS_COMPLETE);
```
Comment (kkrivanka-box) [inline src/elements/content-uploader/ContentUploader.js:942]: nit: this should be called `completedUploads`

**Review by kkrivanka-box — DISMISSED** [review]

File: src/elements/content-uploader/ContentUploader.js (hunk 1 above)
Comment (boxshrutisingh) [reply src/elements/content-uploader/ContentUploader.js:942]: > nit: this should be called `completedUploads`

You mean the prop that we need to pass in -order for calling partial success of files uploaded as success?

Comment (boxshrutisingh) [reply src/elements/content-uploader/ContentUploader.js:942]: @kkrivanka-box Its( areAllItemsFinished) an existing variable which implies -> ```const areAllItemsFinished = items.every(
            uploadItem => uploadItem.status === STATUS_COMPLETE || uploadItem.status === STATUS_ERROR,
        );```
        which is what we want, to check if all the files have either been uploaded or throw error, and nothing is in pending stage.

Comment (boxshrutisingh) [reply src/elements/content-uploader/ContentUploader.js:942]: @kkrivanka-box Its( areAllItemsFinished) an existing variable which implies -> ```const areAllItemsFinished = items.every(
            uploadItem => uploadItem.status === STATUS_COMPLETE || uploadItem.status === STATUS_ERROR,
        );```
        which is what we want, to check if all the files have either been uploaded or throw error, and nothing is in pending stage.

**Review by shahzadaziz — APPROVED** [review]

**Review by kkrivanka-box — APPROVED** [review]

**Pull request merged**